src/app/services/baw-api/base-api.service.ts

```diff
@@ -36,56 +37,66 @@ export abstract class BawApiService<T extends AbstractModel> {
   */
 
   protected userSessionStorage = "baw.client.user";
+  private metaKey = Symbol("meta");
+  private apiRoot: string;
+
+  /**
+   * Handle API collection response
+   * @param response Api Response
+   */
+  private handleCollectionResponse: (response: ApiResponse<T>) => T[];
 
   /**
-   * Handle API response
-   */
-  private handleResponse: (_: ApiResponse<T>) => T[];
-  private handleSingleResponse: (_: ApiResponse<T>) => T;
-
-  // * @param nullResponse True if response body can be empty. If set, next always be called with "true"
-  // {
-
-  // next: <T>(response: ApiResponse<T>) => {
-  //   if (response?.data) {
-  //     next(response.data);
-  //   } else if (!response && nullResponse) {
-  //     // TODO Remove if https://github.com/QutEcoacoustics/baw-server/issues/427 is fixed
-  //     next(true);
-  //   } else {
-  //     error({
-  //       status: apiReturnCodes.unknown,
-  //       message: "No data returned from API"
-  //     } as ApiErrorDetails);
-  //   }
-  // },
-  // error: (err: ApiErrorDetails) => {
-  //   error(err);
-  // }
-  // };
+   * Handle API single model response
+   * @param response Api Response
+   */
+  private handleSingleResponse: (response: ApiResponse<T>) => T;
+
+  /**
+   * Handle API empty response
+   */
+  private handleEmptyResponse(): null {
+    return null;
+  }
 
   constructor(
     protected http: HttpClient,
-    private apiRoot: string,
-    _new: new (_: object) => T) {
+    config: AppConfigService,
+    @Inject(STUB_CLASS_BUILDER) classBuilder: new (_: object) => T
+  ) {
+    this.apiRoot = config.getConfig().environment.apiRoot;
 
-    this.handleSingleResponse = (response: ApiResponse<T>) => {
+    // Create pure functions to prevent rebinding of 'this'
+
+    this.handleCollectionResponse = (response: ApiResponse<T>): T[] => {
       if (response.data instanceof Array) {
-        throw new Error("Received an array of API results when only a single result was expected");
+        return response.data.map(
+          model => new classBuilder(this.addMetadata(model, response.meta))
+        );
+      } else {
+        return [this.handleSingleResponse(response)];
```

Comment:
inline this. This is so simple we don't need to abstract into handle single response

Reply:
I believe you abstracted this to prevent issues with the value of `this` being overwritten

Reply:
No, I mean literally that line:
```suggestion
        return [new classBuilder(this.addMetadata(model, response.meta)];
```